src/components/generic/TimerComps.tsx

```diff
@@ -0,0 +1,289 @@
+            {value}
+        </h1>
+    );
+};
+
+/*
+ * Props for the TimerButton component.
+ */
+type TimerButtonProps = {
+    onClickParam: () => void;
+    timerButtonLabel: string;
+};
+
+/*
+ * Button to controll the workout.
+ */
+export const TimerButton = ({ onClickParam, timerButtonLabel }: TimerButtonProps) => {
+    return <button onClick={onClickParam}> {timerButtonLabel} </button>;
+};
+
+/*
+ * Component for the controllers of the workout.
+ */
+export const ControlsDiv = ({ children }: { children: React.ReactNode }) => {
+    // mx-auto px-6 py-3 flex items-center
+    return <div className="bg-gray-800 absolute bottom-0 right-0 left-0 flex justify-center space-x-4 text-6xl p-5 ">{children}</div>;
+};
+
+/**
+ * Props for the TimerTypeSelect component.
+ */
+type selectTimerProps = {
+    timerTypes: {
+        title: string;
+    }[];
+    selected: number;
+    setSelected: (value: number) => void;
+};
+
+/*
+ * Component to handle the selection of the time of timer to add.
+ */
+export const TimerTypeSelect = ({ timerTypes, selected, setSelected }: selectTimerProps) => {
+    return (
+        <div className="w-64">
+            <label htmlFor="timerType" className="px-4 font-bold">
+                Timer Type
+            </label>
+            <select
+                id="timerType"
+                className="bg-gray-900 text-white rounded-md px-3 py-3 text-sm font-medium"
+                value={selected}
+                onChange={event => {
+                    setSelected(Number.parseInt(event.target.value));
+                }}
+            >
+                {timerTypes.map((timer, index) => (
+                    <option key={timer.title} value={index}>
+                        {timer.title}
+                    </option>
+                ))}
+            </select>
+        </div>
+    );
+};
+
+export const TimerInputsContainer = ({ children }: { children: React.ReactNode }) => {
+    return (
+        <div className="flex flex-wrap w-full">
+            <div className="w-full">{children}</div>
+        </div>
+    );
+};
+
+export type TimerInputComponentProps = {
+    label: string;
+    id: string;
+    defaultValue: number;
+    min: number;
+    max: number;
+    onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+};
+
+export const TimerInputComponent = ({ label, id, defaultValue, min, max, onChange }: TimerInputComponentProps) => {
+    return (
+        <>
+            <div className="flex items-center ">
+                <label className="text-white font-bold mb-1 md:mb-0 whitespace-nowrap mr-2" htmlFor={id}>
+                    {label}
+                </label>
+            </div>
+            <div className="flex items-center col-span-3 w-64">
+                <input
+                    className="appearance-none border-2 border-gray-200 rounded w-full py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-gray-800"
+                    id={id}
```

Comment:
what is this `id` prop used for?

Reply:
The id is so the component is aware of the ID that the values will have the context so it can call the operations like remove, or swap later by providing it back to the context.